src/pseudopeople/utilities.py

```diff
@@ -126,3 +104,48 @@ def add_logging_sink(sink, verbose, colorize=False, serialize=False):
         logger.add(
             sink, colorize=colorize, level="INFO", format=message_format, serialize=serialize
         )
+
+
+def two_d_array_choice(
+    data: pd.Series,
+    options: pd.DataFrame,
+    randomness_stream: RandomnessStream,
+    additional_key: str,
+):
+    """
+    Makes vectorized choice for 2D array options.
+    :param data: pd.Series which should be a subset of options.index
+    :param options: pd.DataFrame where the index is the values of data and columns are available choices.
+    :param randomness_stream: RandomnessStream object
+    :param additional_key: key for randomness_stream
+    :returns: pd.Series with new choices replacing the original values in data.
+    """
+
+    # Change columns to be integers for datawrangling later
+    options.columns = list(range(len(options.columns)))
+    # Get subset of options where we will choose new values
+    data_idx = pd.Index(data.values)
+    options = options.loc[data_idx]
+    # Get number of options per name
+    n = options.count(axis=1)
```

Comment:
You're killing me with this `n` stuff! (I need to then use `!!n` in pdb)

Reply:
Agreed. You should change this variable name to something a bit more descriptive